inference-engine/src/cldnn_engine/cldnn_graph.cpp

```diff
@@ -3339,9 +3339,9 @@ bool CLDNNGraph::IsValidSplitConvMerge(const InferenceEngine::SplitLayer *splitL
     }
 
     auto convLayer1 =
-        as<InferenceEngine::ConvolutionLayer *> (GetNextSingleLayer(splitLayer->outData[0]));
+        dynamic_cast<InferenceEngine::ConvolutionLayer *> (GetNextSingleLayer(splitLayer->outData[0]).get());
```

Comment:
`InferenceEngine:as` performs `dynamic_cast` as well and throws an exception if casted object is `nullptr`. 

@vladimir-paramuzov Do you think we can close this PR or it's valid?

Reply:
@ilya-lavrenov Well, this optimization is a kind of optional, but `as<>` call could throw an exception in case of invalid dynamic_cast inside and fail the whole model processing. But this was fixed long ago and now we sue `tryAs<>` here. So this PR is not actual anymore.